### PR TITLE
Array.prototype.slice() returns this

### LINKED ISSUE #4988: Tuples should be clonable with Array.prototype.slice()
Let's say i define a tuple type like this :
```ts
type MyTuple = [number, number]
```

Now when i do this:
```ts
let a: MyTuple = [1, 2]
let b: MyTuple = a.slice();
```

The compiler complains that `TS2322: Type 'number[]' is not assignable to type '[number, number]'.
 Property '0' is missing in type 'number[]'.`

I think it should work, slice without argument (or slice(0) or even slice(0,2)) should allow to clone a tuple.

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #4988

### CODE REVIEW

**Comment by mhegazy** [comment]
:+1:

File: tests/cases/compiler/library_ArraySlice.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 // Array.prototype.slice can have zero, one, or two arguments
 Array.prototype.slice();
 Array.prototype.slice(0);
-Array.prototype.slice(0, 1);
\ No newline at end of file
+Array.prototype.slice(0, 1);
+
+// Array.prototype.slice returns this with zero arguments, allowing tuple slicing
+let t1: [number, number] = [1, 2];
+let t2: [number, number] = t1.slice();
```
Comment (yuit) [inline tests/cases/compiler/library_ArraySlice.ts]: Could you add these test as well
```ts
let t3:[number, number] = t1.slice(0);
let t3:[number, number] = t1.slice(0,1);
```

Comment (sandersn) [inline tests/cases/compiler/library_ArraySlice.ts]: Sure, although those are negative tests -- only slice() returns `this`.

**Comment by mhegazy** [comment]
is this ready to go?

**Comment by sandersn** [comment]
No, I think it's superceded by my upcoming tuple kind proposal. @DanielRosenwasser and @ahejlsberg pointed out some problems in the discussion on #4988 -- specifically, that `this` implies that additional properties are copied, which is not true with `slice`.

**Comment by mhegazy** [comment]
let's close it then.

**Pull request closed without merging**